examples/step-72/doc/intro.dox

```diff
@@ -73,22 +89,115 @@ obtain
   \nabla \delta u}{\left(1+|\nabla u|^{2}\right)^{\frac{3}{2}}} \nabla u
   \right).
   @f]
-But, this is a sizable expression that is cumbersome both to
-derive and to implement. It is also, in some sense, duplicative: If we
-implement what $F(u)$ is somewhere in the code, then $F'(u,\delta u)$
-is not an independent piece of information but is something that, at
-least in principle, a computer should be able to infer itself.
-This program will show how this can be achieved in practice.
-
-The key point this program tries to make is that this isn't just
-useful for the current nonlinear function $F(u)$, but that the same
-techniques also apply when $F(u)$ may be a much more complicated
-function. A typical example of applications where this happens is in
-<a
-href="https://en.wikipedia.org/wiki/Deformation_(physics)">nonlinear
-deformation</a> and
-<a href="https://en.wikipedia.org/wiki/Elasticity_(physics)">finite
-elasticity</a>, where the nonlinearity of the constitutive equations
-can be very complicated and it can seem essentially impossible to
-correctly derive $F'(u,\delta u)$ by hand, and then implement it
-correctly as well.
+
+So here then is what this program is about: It is about techniques
+that can help us with computing $F'(u,\delta u)$ without having to
+implement it explicitly, either by providing an implementation of
+$F(u)$ or an implementation of $\Psi(u)$. More precisely, we will
+implement three different approaches and compare them in terms of
+run-time but also -- maybe more importantly -- how much human effort
+it takes to implement them:
+- The method used in step-15 to form the Jacobian matrix.
+- Computing the Jacobian matrix from an implementation of the
+  residual $F(u)$, using automatic differentiation.
+- Computing both the residual and Jacobian matrix from an
+  implementation of the energy density $\Psi(u)$.
```

Comment:
Well, as you know we actually implement the discrete $F(U)^{K}$ and $E(U)^{K}$. This motivates my suggesting that we go with $F(u)$ and $E(u)$ in this section.

Reply:
Same here